src/solvers/dgmulti/types.jl

```diff
@@ -76,25 +76,116 @@ function DGMulti(element_type::AbstractElemShape,
   return DG(rd, nothing #= mortar =#, surface_integral, volume_integral)
 end
 
+########################################
+#            DGMultiMesh
+########################################
+
 # now that `DGMulti` is defined, we can define constructors for `DGMultiMesh` which use `dg::DGMulti`
+
+# TODO: DGMulti, v0.5. These constructors which use `rd::RefElemData` are now redundant and can be removed.
+function DGMultiMesh(vertex_coordinates::NTuple{NDIMS, Vector{Tv}}, EToV::Array{Ti,2}, rd::RefElemData;
+                     is_on_boundary = nothing,
+                     periodicity=ntuple(_->false, NDIMS), kwargs...) where {NDIMS, Tv, Ti}
+
+  if haskey(kwargs, :is_periodic)
+    # TODO: DGMulti, v0.5. Remove deprecated keyword
+    Base.depwarn("keyword argument `is_periodic` is now `periodicity`.", :DGMultiMesh)
+    periodicity=kwargs[:is_periodic]
+  end
+
+  md = MeshData(vertex_coordinates, EToV, rd)
+  if NDIMS==1
+    md = StartUpDG.make_periodic(md, periodicity...)
+  else
+    md = StartUpDG.make_periodic(md, periodicity)
+  end
+  boundary_faces = StartUpDG.tag_boundary_faces(md, is_on_boundary)
+  return DGMultiMesh{NDIMS, typeof(rd.elementType), typeof(md), typeof(boundary_faces)}(md, boundary_faces)
+end
+
+function DGMultiMesh(triangulateIO, rd::RefElemData{2, Tri}, boundary_dict::Dict{Symbol, Int})
+
+  vertex_coordinates, EToV = StartUpDG.triangulateIO_to_VXYEToV(triangulateIO)
+  md = MeshData(vertex_coordinates, EToV, rd)
+  boundary_faces = StartUpDG.tag_boundary_faces(triangulateIO, rd, md, boundary_dict)
+  return DGMultiMesh{2, typeof(rd.elementType), typeof(md), typeof(boundary_faces)}(md, boundary_faces)
+end
+
+# TODO: DGMulti, v0.5. Remove deprecated constructor
+@deprecate VertexMappedMesh(args...; kwargs...) DGMultiMesh(args...; kwargs...)
+
+# Mesh types for trait dispatch
+struct Cartesian end
+struct VertexMapped end # where element geometry is determined by vertices.
+struct Curved end
+
+# type parameters for dispatch using `DGMultiMesh`
+struct Affine end # mesh produces constant geometric terms
+struct NonAffine end # mesh produces non-constant geometric terms
```

Comment:
Do you consider these to be implementation details nobody needs to know about or part of the public API?

Reply:
Yes, these are definitely internal at the moment. I might expose them as public API later, but it would be lumped in with the other breaking changes in v0.5.

Reply:
Adding a note on internal use.